### PR TITLE
Autosync edge case with position containing extra FLAT orders

### PR DESCRIPTION
**Pull request by sli-tao**
### Description
Autosync edge case with position containing extra FLAT orders

### Related Issues (JIRA)
[Reference any related issues or tasks that this pull request addresses or closes.]

### Checklist
- [ ] I have tested my changes on testnet.
- [ ] I have updated any necessary documentation.
- [ ] I have added unit tests for my changes (if applicable).
- [ ] If there are breaking changes for validators, I have (or will) notify the community in Discord of the release.

### Reviewer Instructions
[Provide any specific instructions or areas you would like the reviewer to focus on.]

### Definition of Done
- [ ] Code has been reviewed.
- [ ] All checks and tests pass.
- [ ] Documentation is up to date.
- [ ] Approved by at least one reviewer.

### Checklist (for the reviewer)
- [ ] Code follows project conventions.
- [ ] Code is well-documented.
- [ ] Changes are necessary and align with the project's goals.
- [ ] No breaking changes introduced.

### Optional: Deploy Notes
[Any instructions or notes related to deployment, if applicable.]

/cc @mention_reviewer

### CODE REVIEW

File: vali_objects/utils/validator_sync_base.py [hunk 1]
```diff
@@ -194,6 +208,25 @@ def write_modifications(self, position_to_sync_status, stats):
...
+        """
+        if p2 is None:
+            return p1
+
+        self.global_stats['n_positions_closed_duplicate_opens_for_trade_pair'] += 1
+
+        # if p2 is an older position, we close it and return p1 as the newest open position.
+        if p2.open_ms < p1.open_ms:
+            p2.close_out_position(TimeUtil.now_in_millis())
+            self.position_manager.save_miner_position_to_disk(p2, delete_open_position_if_exists=True)
```
Comment (jbonilla-tao) [inline vali_objects/utils/validator_sync_base.py:223]: delete_open_position_if_exists=False

File: vali_objects/utils/validator_sync_base.py [hunk 2]
```diff
@@ -194,6 +208,25 @@ def write_modifications(self, position_to_sync_status, stats):
...
+            return p1
+        else:
+            p1.close_out_position(TimeUtil.now_in_millis())
+            self.position_manager.save_miner_position_to_disk(p1, delete_open_position_if_exists=True)
```
Comment (jbonilla-tao) [inline vali_objects/utils/validator_sync_base.py:227]: delete_open_position_if_exists=False

File: vali_objects/utils/position_manager.py [hunk 3]
```diff
@@ -993,6 +993,15 @@ def save_miner_position_to_disk(self, position: Position, delete_open_position_i
 
         ValiBkpUtils.write_file(miner_dir + position.position_uuid, position)
 
+    def overwrite_position_on_disk(self, position: Position) -> None:
+        # delete the position from disk. Try the open position dir and the closed position dir
+        self.delete_position_from_disk(position, check_open_and_closed_dirs=True)
+        miner_dir = ValiBkpUtils.get_partitioned_miner_positions_dir(position.miner_hotkey,
+                                                                     position.trade_pair.trade_pair_id,
+                                                                     order_status=position.is_open_position,
```
Comment (sli-tao) [inline vali_objects/utils/position_manager.py:1001]: Should this be an OrderStatus enum?

Comment (sli-tao) [reply vali_objects/utils/position_manager.py:1001]: order_status=OrderStatus.OPEN if position.is_open_position else OrderStatus.CLOSED

**Review by jbonilla-tao — APPROVED** [review]

**Pull request merged**